oak-run/src/test/resources/org/apache/jackrabbit/oak/index/merge/merge-override-tags-type.json

```diff
@@ -0,0 +1,53 @@
+{
+  "build": {
+    "/oak:index/lucene-2": {
+      "jcr:primaryType": "nam:oak:QueryIndexDefinition",
+      ":version": 2,
+      "tags": ["similarity", "asset", "fragments"],
+      "type": "elasticsearch",
+      "async": "async",
+      "reindex": false,
+      "reindexCount": 1
+    }
+  },
+
+  "run": {
+    "/oak:index/lucene-1": {
+      "jcr:primaryType": "nam:oak:QueryIndexDefinition",
+      ":version": 2,
+      "tags": ["similarity", "asset"],
+      "type": "disabled",
+      "async": "async",
+      "reindex": false,
+      "reindexCount": 1
+    },
+    "/oak:index/lucene-1-custom-1": {
+      "jcr:primaryType": "nam:oak:QueryIndexDefinition",
+      ":version": 2,
+      "tags": "custom",
+      "type": "lucene",
+      "async": "async",
+      "reindex": false,
+      "reindexCount": 1
+    }
+  },
+
+  "expected": {
+    "/oak:index/lucene-2": {
+      "jcr:primaryType": "nam:oak:QueryIndexDefinition",
+      ":version": 2,
+      "tags": ["similarity", "asset", "fragments"],
+      "type": "elasticsearch",
+      "async": "async",
+      "reindex": false,
+      "reindexCount": 1
+    },
+    "/oak:index/lucene-2-custom-1": {
+      "jcr:primaryType": "nam:oak:QueryIndexDefinition",
+      "tags": ["asset", "custom", "fragments"],
```

Comment:
I don't fully understand this:
* lucene-1 -> `"tags": ["similarity", "asset"]`
* lucene-1-custom-1 -> `"tags": "custom"` (tags have been completely overwritten)
* lucene-2 -> ` "tags": ["similarity", "asset", "fragments"]` (fragments was added compared to the previous OOTB)
* lucene-2-custom-1 -> "tags": ["asset", "custom", "fragments"] 

Shouldn't the merged index have `"tags": ["custom", "fragments"]`?

Reply:
You are right, this is incorrect... actually it should be:
* lucene-1: `"tags": ["asset", "custom", "fragments", "similarity"]`

Basically, the customer should not be able to remove a tag.

I will fix this and add more unit tests.

Reply:
Even worse, "async" merging was wrong... Fixed.